src/test/java/com/learning/selenium/app/Application.java

```diff
@@ -0,0 +1,47 @@
+package com.learning.selenium.app;
+
+import com.learning.selenium.pages.*;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class Application {
+
+    public WebDriver driver;
+    private PasteMainPage pasteMainPage;
+    private GoogleCloudMainPage googleCloudMainPage;
+    private GoogleCloudHeader googleCloudHeader;
+    private GoogleCloudSearchResultsPage googleCloudSearchResultsPage;
+    private GoogleCloudPricingCalcPage googleCloudPricingCalcPage;
+
+    public Application() {
+        driver = new ChromeDriver();
+        pasteMainPage = new PasteMainPage(driver);
+        googleCloudMainPage = new GoogleCloudMainPage(driver);
+        googleCloudHeader = new GoogleCloudHeader(driver);
+        googleCloudSearchResultsPage = new GoogleCloudSearchResultsPage(driver);
+        googleCloudPricingCalcPage = new GoogleCloudPricingCalcPage(driver);
+    }
+
+    public void stop() {
+        driver.quit();
```

Comment:
I think this should be done somewhere in the base-test logic

Reply:
Yes, but the idea was not to do it in the base-test logic, but to split it into two levels:
- At the first level there is BaseTest, which, before each test execution, initializes an object with the logic of interaction with the application (Application) and destroys it after each execution. He also acts as a parent to the tests.
And there are classes with tests
- At the second level, classes of interaction with the application. Application is a single point of access to these classes and all the details of using selenium (driver, etc.) are hidden in it